datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -284,9 +288,7 @@ fn _date_trunc(
     };
 
     // convert to nanoseconds
-    let Some(nano) = (f)(Some(value * scale))? else {
-        return Ok(None);
-    };
+    let nano = date_trunc_coarse(granularity, scale * value)?;
```

Comment:
There is no need for a closure, and the function can be called directly which I think makes things easier to understand